tests/api/v2/datadog/api_logs_archives_test.go

```diff
@@ -0,0 +1,160 @@
+	gock.New(URL).Get(fmt.Sprintf("/api/v2/logs/config/archives/%s", id)).Reply(200).JSON(outputArchive)
+	result, httpresp, err := client.LogsArchivesApi.GetLogsArchive(ctx, id).Execute()
+	assert.Equal(httpresp.StatusCode, 200)
+	checkS3Archive(t, *outputArchive.Get().Data)
+	assert.Equal(result, *outputArchive.Get())
+	defer gock.Off()
+}
+
+func TestLogsArchivesDelete(t *testing.T) {
+	if tests.GetRecording() == tests.ModeReplaying {
+		t.Skip("This test case does not support reply from recording")
+	}
+	ctx, finish := WithClient(WithFakeAuth(context.Background()), t)
+	defer finish()
+	assert := tests.Assert(ctx, t)
+	id := "XVlBzgbaiC"
+	action := "deleteById"
+	archiveType := "s3"
+	client := Client(ctx)
+	URL, err := client.GetConfig().ServerURLWithContext(ctx, fmt.Sprintf("LogsArchive.%s.%s", action, archiveType))
+	assert.NoError(err)
+	gock.New(URL).Delete(fmt.Sprintf("/api/v2/logs/config/archives/%s", id)).Reply(204)
+	httpresp, err := client.LogsArchivesApi.DeleteLogsArchive(ctx, id).Execute()
+	assert.NoError(err)
+	assert.Equal(httpresp.StatusCode, 204)
+	defer gock.Off()
+}
+
+func TestLogsArchivesGetAll(t *testing.T) {
+	if tests.GetRecording() == tests.ModeReplaying {
+		t.Skip("This test case does not support reply from recording")
+	}
+	ctx, finish := WithClient(WithFakeAuth(context.Background()), t)
+	defer finish()
+	assert := tests.Assert(ctx, t)
+	action := "getall"
+	archiveType := "s3"
+	outputArchives := datadog.NullableLogsArchives{}
+	outputArchives.UnmarshalJSON([]byte(readFixture(t, fmt.Sprintf("fixtures/logs/archives/%s/out/%s.json", archiveType, action))))
+	URL, err := Client(ctx).GetConfig().ServerURLWithContext(ctx, fmt.Sprintf("LogsArchive.%s.%s", action, archiveType))
+	assert.NoError(err)
+	gock.New(URL).Get("/api/v2/logs/config/archives/").Reply(200).JSON(outputArchives)
+	assert.True(len(*outputArchives.Get().Data) > 0)
+	checkS3Archive(t, (*outputArchives.Get().Data)[0])
+	defer gock.Off()
+}
+
+func checkS3Archive(t *testing.T, outputArchive datadog.LogsArchiveDefinition) {
+	assert.Equal(t, outputArchive.Type, "archives")
+	destination := outputArchive.Attributes.Destination.Get().LogsArchiveDestinationS3
+	assert.Equal(t, destination.Type, datadog.LOGSARCHIVEDESTINATIONS3TYPE_S3)
+	assert.Equal(t, destination.Integration.AccountId, "711111111111")
+	assert.Equal(t, destination.Integration.RoleName, "DatadogGoClientTestIntegrationRole")
+	assert.Equal(t, destination.Path, datadog.PtrString("/path/blou"))
+	assert.Equal(t, destination.Bucket, "dd-logs-test-datadog-api-client-go")
+	assert.Equal(t, outputArchive.Attributes.Name, "datadog-api-client-go Tests Archive")
+	assert.Equal(t, outputArchive.Attributes.Query, "source:tata")
+	assert.Equal(t, outputArchive.Id, datadog.PtrString("XVlBzgbaiC"))
+}
+
+func archivesAreEqual(t *testing.T, archiveRequest datadog.LogsArchiveCreateRequestAttributes, createdArchive datadog.LogsArchiveAttributes) {
+	assert.Equal(t, archiveRequest.Name, createdArchive.Name)
```

Comment:
Please use instance of `tests.Assert(ctx, t)` so we can have nice trace information.